scripts/nice-history/nice-history.js

```diff
@@ -0,0 +1,220 @@
+// ==UserScript==
+// @name         Nice history
+// @match        https://pokefarm.com/*
+// @icon         https://pokefarm.com/favicon.ico
+// @grant        none
+// ==/UserScript==
+
+/*
+Author: Mirzam
+Keeps track of Nices recieved, so you can easily see who sent them and when!
+Nices are ONLY stored locally per device, since the main intent is to be able to view a missed nice.
+Report bugs and issues in my main code sharing thread: https://pfq.link/~-HvB
+
+For testing: Manually generate a test nice
+Test nices cannot be dismissed normally, but should be detected by the script
+
+username = 'USERNAME';
+testNice = document.createElement('div');
+testNice.className = 'nice_toast';
+testNice.innerHTML = '<div class="nice_star1"></div><div class="nice_star2"></div><div class="nice_star3"></div><a href="#" class="nice_close"></a>'+username+'<br>gave you a Nice!';
+document.body.append(testNice);
+
+*/
+
+(function() {
+  let userID = null;
+  let recentNices = [];
+
+  // You should always call this instead of accessing userID directly
+  const getUserID = () => {
+    // Detect the current user to set cookie for
+    if(!userID) {
+      if(document.getElementById('core')) {
+        userID = document.getElementById('core').attributes['data-user'].value;
+        console.log('Nice watcher detected user: '+userID);
+      }
+      else {
+        console.error('Nice watcher failed to detect user');
+      }
+    }
+    return userID;
+  }
+
+  const niceWatch = () => {
+    // Detect if a nice has happened
+    let niceEls = document.querySelectorAll('body > .nice_toast');
+    if(niceEls) {
+      niceEls.forEach(niceEl => {
+          // don't detect nices in the display window
+          let nice = niceEl.innerHTML;
+          console.warn('Nice detected!');
+          console.log(nice);
+          storeNice(nice);
+      });
+    }
+  }
+
+  const storeNice = (nice) => {
```

Comment:
Rather than passing in `innerHTML` and parsing it manually, I feel like it would be much simpler to pass in the toast itself, and maybe `querySelector` it for the userlink. This gives you the URL, name and even prefixed icons for Hypermode/New Farmer, which could very much improve both the code's reliability, and the UI.